caikit_nlp/modules/text_generation/text_generation_tgis.py

```diff
@@ -109,6 +105,14 @@ def __del__(self):
         if self._model_loaded and self.tgis_backend:
             self.tgis_backend.unload_model(self.model_name)
 
+    @cached_property
+    def _client(self):
+        # Configure the internal client
+        # NOTE: This is made optional for the cases where we do not need to execute `.run` function
+        # for example, bootstrapping a model to caikit format and saving.
+        if hasattr(self, "tgis_backend") and self.tgis_backend:
+            return self.tgis_backend.get_client(self.model_name)
```

Comment:
Added a `_tgis_backend` property which is guaranteed to exist. Allows us to do a simple `if self._tgis_backend` instead of using `getattr()` which breaks intellisense/autocomplete in IDEs.